batch2/batch/worker.py

```diff
@@ -514,24 +525,37 @@ def id(self):
     #   name: str,
     #   batch_id: int,
     #   job_id: int,
+    #   attempt_id: int,
     #   user: str,
     #   state: str, (pending, initializing, running, succeeded, error, failed)
     #   error: str, (optional)
     #   container_statuses: [Container.status]
+    #   start_time: float
+    #   end_time: float
     # }
     async def status(self):
         status = {
             'worker': NAME,
             'batch_id': self.batch_id,
             'job_id': self.job_spec['job_id'],
+            'attempt_id': self.job_spec['attempt_id'],
             'user': self.user,
-            'state': self.state
+            'state': self.state,
         }
         if self.error:
             status['error'] = self.error
-        status['container_statuses'] = {
+
+        cstatuses = {
             name: await c.status() for name, c in self.containers.items()
         }
+        status['container_statuses'] = cstatuses
+
+        main_timings = cstatuses['main']['timing']
+        main_duration = main_timings['starting']['duration'] + main_timings['running']['duration']
+
+        status['start_time'] = self.start_time
+        status['end_time'] = self.start_time + main_duration
```

Comment:
Main may not have timing, so this isn't safe.  I'm not sure what you're calculating here.  My intention was that the timing 'runtime' was what we wanted to report (it measures the time spent in the cpu semaphore), so you should just be able to grab main['timing']['runtime']['start_time'/'end_time'].

Reply:
My concern was that acquiring the semaphore occurs after starting the clock for the runtime. I think we can just change the order of those two statements and then I can do the easier thing you suggested.

Reply:
Oh, you're totally right.  The timing should be inside the semaphore, that's a total mistake.